### PR TITLE
Feature/user dropdown

### PR DESCRIPTION
**Pull request by TimCreasman**
Add the user dropdown to the client. Add support for the Google Directory API and user dropdown on the server.

### CODE REVIEW

File: client/src/app/services/asset.service.js [hunk 1]
```diff
@@ -295,21 +295,23 @@
 
                 return defer.promise;
             },
+
+
```
Comment (tinoweiss) [inline client/src/app/services/asset.service.js]: Unnecessary newline (only 1 needed)

File: client/src/app/services/asset.service.js [hunk 2]
```diff
@@ -295,21 +295,23 @@
...
             /**
              * Checks out asset associated with the assetID until the specified return date.
              * @param assetID -> id of the asset involved
              * @param returnDate -> date chosen to return the asset by
+             * @param user -> (optional) object of the user
              * @returns {*} -> Promise object whose data is the udpated asset details
              */
-            checkoutAsset: function(assetID, returnDate, checkoutEmail) {
+            checkoutAsset: function(assetID, returnDate) {
```
Comment (tinoweiss) [inline client/src/app/services/asset.service.js]: Comments don't align with function declaration.

File: client/src/app/services/asset.service.js [hunk 3]
```diff
@@ -326,93 +328,113 @@
...
+                    });
 
+                //return the updated asset object
+                return defer.promise;
+            },
 
-                            } else if (status === 202) {
-                                defer.resolve(ret);
-                            }
+            checkoutAssetForUser: function(assetID, returnDate, userInfo) {
```
Comment (tinoweiss) [inline client/src/app/services/asset.service.js]: JSDoc needed

**Comment by tinoweiss** [comment]
Please make the suggested changes above. I'll review the functionality of the feature later before merging in the PR.

**Comment by tinoweiss** [comment]
The following functionality needs to be added:
1. Use the Google Directory
2. Use the CheckIT user database
3. Manually search for user by email address.

Options 2 and 3 will allow other ways to search for a user or check out a device for another user if the Google Directory call fails.

**Comment by tinoweiss** [comment]
Bug when checking out for someone else: if user does not select an entry from the dropdown list, a 'TypeError' occurs after clicking Apply to check the device out. This can occur if the user's name is typed in and not selected from the dropdown, the user's email is typed in and not selected from the dropdown, or another string is entered without selecting an entry from the dropdown.
screenshot 2017-07-17 10 09 02

**Comment by tinoweiss** [comment]
Error modal when "Check out for someone else" input is invalid is missing content.
screenshot 2017-07-17 14 57 46

**Comment by TimCreasman** [comment]
"Error modal when "Check out for someone else" input is invalid is missing content."
This is not a bug with the user dropdown code. It exists within the modalService code.

**Pull request merged**